schema.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * The top-level interface of a custom-elements.json file.
+ * 
+ * custom-elements.json documents all the elements in a single npm package,
+ * across all modules within the package. Elements may be exported from multiple
+ * modules with re-exports, but as a rule, elements in this file should be
+ * included once in the "canonical" module that they're exported from.
+ */
+export interface PackageDoc {
+  version: string;
+
+  /**
+   * An array of the modules this package contains.
+   */
+  modules: Array<ModuleDoc>;
+}
+
+export interface ModuleDoc {
+  path: string;
+
+  /**
+   * A markdown summary suitable for display in a listing.
+   */
+  summary?: string;
+
+  /**
+   * A markdown description of the module.
+   */
+  description?: string;
+
+  exports?: Array<ExportDoc>;
+}
+
+export type ExportDoc = ClassDoc|FunctionDoc|VariableDoc;
+
+/**
+ * A reference to an export of a module.
+ * 
+ * All references are required to be publically accessible, so the canonical
+ * representation of a refernce it the export it's available from.
+ */
+export interface Reference {
+  name: string;
+  package?: string;
+  module?: string;
+}
+
+export interface CustomElementDoc extends ClassDoc {
+  tagName: string;
+  /**
+   * The attributes that this element is known to understand.
+   */
+  attributes?: AttributeDoc[];
+
+  /** The events that this element fires. */
+  events?: EventDoc[];
+
+  /**
+   * The shadow dom content slots that this element accepts.
+   */
+  slots?: SlotDoc[];
+
```

Comment:
misses css custom properties right?

Reply:
they are now exports of the class? seems kinda like a stretch as to have css variables as part of javascript class exports?

also seems same problem as with not having flattened properties... tough to work and imho also to understand 🙈

Reply:
I'll add the styles section. This is in an interface that extends a ClassDoc, so ti has everything classes do plus custom element specifics.